src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1071,13 +1103,16 @@ protected void reportEnvironment(TaskListener listener) throws IOException, Inte
         connection.exec("set",listener.getLogger());
     }
 
+    @NonNull
     private String checkJavaVersion(TaskListener listener, String javaCommand) throws IOException, InterruptedException {
         listener.getLogger().println(Messages.SSHLauncher_CheckingDefaultJava(getTimestamp(),javaCommand));
         StringWriter output = new StringWriter();   // record output from Java
 
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         connection.exec(javaCommand + " "+getJvmOptions() + " -version",out);
-        BufferedReader r = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(out.toByteArray())));
+        //TODO: Seems we need to retrieve the encoding from the connection destination
+        BufferedReader r = new BufferedReader(new InputStreamReader(
```

Comment:
not with a byte array stream as there are no external resources to leak... but it could cause complaints of a resource leak...